### PR TITLE
Change all `port` parameters to use `Stdlib::Port`

### PR DESCRIPTION
**Pull request by bastelfreak**
Instead of having a mix of strings and integers, all port parameters
should be using `Stdlib::Port`.

I've also changed some `String`s to `Stdlib::Host`, but none of these
should be breaking changes.

### CODE REVIEW

**Comment by bastelfreak** [comment]
Replaces https://github.com/voxpupuli/puppet-collectd/pull/870

I created a new PR because I wasn't allowed to push to your branch.

**Comment by bastelfreak** [comment]
Hey people. Would be nice if someone can review and approve this. I would like to wait with the merge until we've some other backwards incompatible changes.

File: manifests/plugin/ovs_stats.pp [hunk 1]
```diff
@@ -27,7 +27,7 @@
 #  Defaults to 'collectd-ovs_stats'
 #
 # [*port*]
-#  TCP-port to connect to. Either a service name or a port number may be given.
+#  TCP-port to connect to.
```
Comment (igalic) [inline manifests/plugin/ovs_stats.pp]: so this pr is compatibility break

File: manifests/plugin/postgresql/database.pp [hunk 2]
```diff
@@ -2,9 +2,9 @@
 # useful if you have multiple instances of different version of pg
 define collectd::plugin::postgresql::database (
   $ensure       = 'present',
```
Comment (igalic) [inline manifests/plugin/postgresql/database.pp]: this could use more types ;)

**Pull request merged**